Features/ether wallet

DenysB-DevBrother opened the pull request.

Commit:
add ether wallet

Commit:
fix saving wallet

Commit:
fix

Commit:
Update TransactionHistory.tsx

Commit:
Update Wallet screen navigation back button functionality

Commit:
update

Commit:
Add zero-based indexing for wallet functions

Commit:
Rename generateKeyAndAddress to generatePrivateKey

Commit:
Use wallet.address instead of privateKey for transfer requests

Commit:
Fix lint issues

Commit:
Revert "Fix lint issues"

This reverts commit a31e8508e3b5b2368c0512db1e591ac785ed1724.

IhorR-DevBrother commented:
bug:
- create second wallet
- press back button
- new wallet is not present on list

IhorR-DevBrother commented:
this is completely different to what we have in designs. 
![image](https://github.com/user-attachments/assets/3f73b138-000c-457d-9543-57219b961da2)

Commit:
Update column names in TransactionHistory.tsx

Commit:
Update wallets state management in Provider component

Commit:
Fix import paths and rename wallet-related files - Update imports to
reflect new file locations - Rename TransactionHistory to components
directory - Adjust Wallet and Wallets screen references

Commit:
Update app configuration for Bluesky web app and adjust Android SDK versions

Commit:
Refactor imports and update function parameters across multiple components

IhorR-DevBrother commented on src/view/com/modals/CreateWallet.tsx:
```diff
@@ -24,22 +29,33 @@ export function Component({}: Props) {
   const createWallet = useCallback(async () => {
-    const wallet = generateKeyAndAddress()
-    const saveRes = await saveWalletToFS(wallet)
+    const privateKey = generatePrivateKey() as Uint8Array
```
just change generatePrivateKey return type. it can never return hex anyway

IhorR-DevBrother commented on src/lib/wallet.ts:
```diff
@@ -59,43 +44,61 @@ export function generateKeyAndAddress(): Wallet {
+        'application/x-pem-file',
+      )
+    case WalletType.ETHERIUM:
+      return await saveToDevice(`${address}.key`, key, 'application/x-pem-file')
```
this doesnt typecheck because key is not a string

IhorR-DevBrother commented on src/lib/wallet.ts:
```diff
@@ -59,43 +44,61 @@ export function generateKeyAndAddress(): Wallet {
-    JSON.stringify(json, undefined, 4),
-    'application/json',
+export function fetchF1r3SkyWalletState(
+  agent: AtpAgent,
```
not used

IhorR-DevBrother committed:
fix

DenysB-DevBrother commented on src/screens/Wallets/Wallet.tsx:
```diff
@@ -1,121 +1,254 @@
+          <Trans>LOADING...</Trans>
+        </Text>
+      </View>
+    )
```
<img width="480" height="548" alt="image" src="https://github.com/user-attachments/assets/d5f884bd-0a2a-4afe-9f03-0c2754e4f0d9" />

Commit:
Update WalletAddress component to display more address characters

Improve loading screen UI in Wallet screen and update wallet header with
new components.
